### PR TITLE
Document DateTime.diff/2 leaps

### PR DESCRIPTION
**Pull request by ericmj**

### CODE REVIEW

File: lib/elixir/lib/calendar/datetime.ex [hunk 1]
```diff
@@ -665,6 +665,8 @@ defmodule DateTime do
   Subtracts `datetime2` from `datetime1`.
 
   The answer can be returned in any `unit` available from `t:System.time_unit/0`.
+  
+  Leaps in time are not considered, such as leap seconds or leap days.
```
Comment (michalmuskala) [inline lib/elixir/lib/calendar/datetime.ex]: Are you sure about leap days? The `to_iso_days` conversion takes leap days into account.

Comment (josevalim) [reply lib/elixir/lib/calendar/datetime.ex]: Only leap seconds are not taken into account. leap days always and everywhere.

**Comment by ericmj** [comment]
Fixed based on feedback.

**Comment by wojtekmach** [comment]
Should we have the same comment in NaiveDateTime.diff/2?

**Pull request merged**